modules/programs/lf.nix

```diff
@@ -68,27 +20,18 @@ in {
       };
 
       settings = mkOption {
-        type = lfSettingsType;
+        type = types.attrs;
```

Comment:
`types.attrs` is deprecated. We should use a more precise type that covers everything we need, like `let val = oneOf [ bool int str ]; in attrsOf (either val (listOf val))`.

Reply:
Done. Interestingly, `oneOf [ int str (listOf int) (listOf str) ]` doesn't work. You need to do `oneOf [ int str (listOf (either int str)) ]`, I wonder why.

Reply:
Because the NixOS module system is stupid. `(listOf int).check ["a" "b"]` returns `true` because `check` is non-recursive, so that branch is selected and then `int.check "a"` fails.